migrations/2024-11-21-195004_add_report_count/up.sql

```diff
@@ -0,0 +1,79 @@
+-- Adding report_count and unresolved_report_count
+-- to the post and comment aggregate tables
+ALTER TABLE post_aggregates
+    ADD COLUMN report_count bigint NOT NULL DEFAULT 0,
+    ADD COLUMN unresolved_report_count bigint NOT NULL DEFAULT 0;
+
+ALTER TABLE comment_aggregates
+    ADD COLUMN report_count bigint NOT NULL DEFAULT 0,
+    ADD COLUMN unresolved_report_count bigint NOT NULL DEFAULT 0;
+
+-- Update the historical counts
+-- Posts
+UPDATE
+    post_aggregates AS a
+SET
+    report_count = cnt.count
+FROM (
+    SELECT
+        post_id,
+        count(*) AS count
+    FROM
+        post_report
+    GROUP BY
+        post_id) cnt
+WHERE
+    a.post_id = cnt.post_id;
+
+-- The unresolved
+UPDATE
+    post_aggregates AS a
+SET
+    unresolved_report_count = cnt.count
+FROM (
+    SELECT
+        post_id,
+        count(*) AS count
+    FROM
+        post_report
+    WHERE
+        resolved = 'f'
+    GROUP BY
+        post_id) cnt
+WHERE
+    a.post_id = cnt.post_id;
+
+-- Comments
+UPDATE
+    comment_aggregates AS a
+SET
+    report_count = cnt.count
+FROM (
+    SELECT
+        comment_id,
+        count(*) AS count
+    FROM
+        comment_report
+    GROUP BY
+        comment_id) cnt
+WHERE
+    a.comment_id = cnt.comment_id;
+
+-- The unresolved
+UPDATE
+    comment_aggregates AS a
+SET
+    unresolved_report_count = cnt.count
+FROM (
+    SELECT
+        comment_id,
+        count(*) AS count
+    FROM
+        comment_report
+    WHERE
+        resolved = 'f'
+    GROUP BY
+        comment_id) cnt
+WHERE
+    a.comment_id = cnt.comment_id;
```

Comment:
I would limit this to posts/comments from the last month or so. Before that reports should already be taken care of, so it would be a lot of migration time for no benefit.

Reply:
Its only really querying the report tables. I just tested them both on lemmy.ml's production DB locally, and they're < 10ms.

Reply:
Interesting I would have thought that its similarly slow as the controversial rank migration as they both have to go through all posts.

Reply:
controversial rank had to do some exponential math in a postgres procedure for every single post (> 700k), this only needs to query the post_report tables which had <3k rows, and even faster with the group by.